src/libsyntax_pos/symbol.rs

```diff
@@ -708,7 +728,12 @@ impl Encodable for LocalInternedString {
     }
 }
 
-/// Represents a string stored in the string interner.
+/// A thin wrapper around `Symbol`. It has two main differences to `Symbol`.
+/// - Its implementations of `Hash`, `PartialOrd` and `Ord` work with the
+///   string chars rather than the symbol integer. This is useful when
+///   hash stability is required across compile sessions, or a guaranteed sort
+///   ordering is required.
+/// - It is guaranteed to not be a gensym symbol.
```

Comment:
This guarantee doesn't make much sense, `InternedString` represents *string contents*, internal `Symbol` is an implementation detail.
`gensym_symbol.as_interned_str()` just loses its gensymness in the process of conversion because it's not a part of string contents (which causes issues, https://github.com/rust-lang/rust/issues/49300).

Reply:
Ok, I will fix.